### PR TITLE
chore(medusa): Improve store list products

### PR DESCRIPTION
**Pull request by adrien2p**
WIP (~46.5 percents. by playing with references. since we are calling set* methods prefer assigning over copying and therefore avoid some copies of potentially a certain amount of heavy objects. the previous last point allows to use concurrency as assigning in that case on the reference allows us to be concurrent as the previous values are not necessary for the next one, which previously would have required to re copy again to obtain the same result, which therefore also avoid plenty of copy)

As an example, the store integration tests for the product list with a cart id went from 385ms to 206ms for the list request to be performed (reproduced multiple times). Of course the previous times are not to take as a truth but just as a display of an improvement.

### CODE REVIEW

**Comment by adrien2p** [comment]
I added some todo to not overload this pr as it will be part of the DD

**Comment by adrien2p** [comment]
@olivermrbl can I merge this one?

**Comment by olivermrbl** [comment]
@adrien2p I'll give this a review today 👍

File: packages/medusa/src/api/routes/store/products/list-products.ts [hunk 1]
```diff
@@ -212,37 +213,50 @@ export default async (req, res) => {
...
+  // Create a new reference just for naming purpose
+  const computedProducts = rawProducts
 
-  const products = await productVariantInventoryService.setProductAvailability(
-    pricedProducts,
-    filterableFields.sales_channel_id
-  )
+  // We can run them concurrently as the new properties are assigned to the references
+  // of the appropriate entity
+  await Promise.all([
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/store/products/list-products.ts:244]: praise: This is great!

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**